cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/SourceControlOperationRunner.java

```diff
@@ -36,4 +37,11 @@ public interface SourceControlOperationRunner {
   PushAppResponse push(NamespaceId namespace, RepositoryConfig repoConfig,
                        ApplicationDetail appToPush, CommitMeta commitDetails)
     throws PushFailureException, NoChangesToPushException, AuthenticationConfigException;
+
+  /**
+   * Pull the requested application from linked repository.
+   * @param appRef The {@link ApplicationReference} of the application to pull from
+   * @return {@link PullAppResponse} of the requested application.
+   */
+  PullAppResponse pull(ApplicationReference appRef);
```

Comment:
Similar to push, this will need a repoConfig as well.

Reply:
But once CDAP-20396 is in, we don't need to pass it in?

Reply:
Go it, theres a dependency b/w the PRs. I'm working on https://github.com/cdapio/cdap/pull/14928, I'll make the change to the signature in my PR if CDAP-20396 goes in first. 
In case my PR is submitted first, you'll need to make the change in pull also.